python/pyarrow/array.pxi

```diff
@@ -232,6 +232,11 @@ def array(object obj, type=None, mask=None, size=None, from_pandas=None,
     else:
         c_from_pandas = from_pandas
 
+    if isinstance(obj, Array):
+        if type is not None and not obj.type.equals(type):
+            obj = obj.cast(type)
```

Comment:
We probably should pass through the `safe` and `memory_pool` keywords to `cast` (`safe` can also easily be tested)

Reply:
Hm, [pc.cast](https://arrow.apache.org/docs/python/generated/pyarrow.compute.cast.html) doesn't accept the `memory_pool` keyword argument?

Reply:
To update the thread: as we discussed I will add the change to the `cast()` method to accept `memory_pool` argument as a part of this PR.